labs/db-iterator/src/iterator.spec.ts

```diff
@@ -0,0 +1,123 @@
+import anyTest, { TestInterface } from 'ava';
+import iterateFind from '.';
+// TODO(ariels): Change db-client to allow starting a client late,
+//     *without* environment variables.
+import { DB, Q } from '@reshuffle/db/dist/db';
+import { mkdtemp } from 'fs';
+import rmrf from 'rmfr';
+import { tmpdir } from 'os';
+import { promisify } from 'util';
+import * as path from 'path';
+import { Handler } from '@reshuffle/leveldb-server';
+import { DBRouter } from '@reshuffle/interfaces-koa-server';
+import Koa from 'koa';
+import KoaRouter from 'koa-router';
+import { AddressInfo } from 'net';
+
+// TODO(ariels): Extract; used also in db-client.
+interface Context {
+  db: DB;
+  dbDir: string;
+}
+
+const test = anyTest as TestInterface<Context>;
+
+const app = new Koa();
+const router = new KoaRouter();
+app.use(router.routes());
+app.use(router.allowedMethods());
+const server = app.listen(0, '127.0.0.1');
+const port = new Promise<number>((res) => {
+  server.once('listening', () => {
+    res((server.address() as AddressInfo).port);
+  });
+});
+
+let count = 0;
+
+test.beforeEach(async (t) => {
+  const dbDir = await promisify(mkdtemp)(path.join(tmpdir(), 'test-state-'), 'utf8');
+  const handler = new Handler(`${dbDir}/root.db`, undefined, (err) => t.fail(err && err.toString()));
+  const dbRouter = new DBRouter(handler);
+  const title = `${++count}`;
+  router.use(`/${title}`, dbRouter.koaRouter.routes(), dbRouter.koaRouter.allowedMethods());
+
+  const ctx = {
+    appId: `testing:${title}`,
+    appEnv: 'default',
+    collection: 'test',
+    auth: {},
+  };
+  const db = new DB(`http://localhost:${await port}/${title}`, ctx, { timeoutMs: 1000 });
+
+  t.context = {
+    db,
+    dbDir,
+  };
+});
+
+test.afterEach(async (t) => {
+  await rmrf(t.context.dbDir);
+});
+
+async function iteratorToArray<T>(it: AsyncIterableIterator<T>): Promise<T[]> {
```

Comment:
It might makes sense to export this (trivial) function too. Or in documentation suggest using https://www.npmjs.com/package/streaming-iterables for utilities like `collect` or `reduce`.

Reply:
Doing the latter.  I don't want to provide utilities to simplify this, it's an antipattern when using an async iterable.  It makes sense in this test, but is less useful generally.

Also you're right: we should gauge interest in different interfaces by providing `db-stream` and maybe even `db-observable`.